barretenberg/cpp/src/barretenberg/ultra_vanilla_client_ivc/mock_circuit_producer.hpp

```diff
@@ -0,0 +1,163 @@
+
+#include "barretenberg/common/op_count.hpp"
+#include "barretenberg/goblin/mock_circuits.hpp"
+#include "barretenberg/stdlib_circuit_builders/ultra_circuit_builder.hpp"
+#include "barretenberg/ultra_honk/ultra_verifier.hpp"
+#include "barretenberg/ultra_vanilla_client_ivc/ultra_vanilla_client_ivc.hpp"
+
+using namespace bb;
+
+namespace {
+
+/**
+ * @brief Test utility for coordinating passing of databus data between mocked private function execution circuits
+ * @details Facilitates testing of the databus consistency checks that establish the correct passing of databus data
+ * between circuits. Generates arbitrary return data for each app/kernel. Sets the kernel calldata and
+ * secondary_calldata based respectively on the previous kernel return data and app return data.
+ */
+class MockDatabusProducer {
```

Comment:
This is just copied from its other location yeah? maybe you meant to do something with it but for now it should just be deleted?